### DISCUSSION TITLE
BME280 Sensor - Its possible to convert humidity to float?

### DISCUSSION THREAD

**Discussion by pierregraciano** [micropython/micropython#13757] [General]
Im trying to calibrate my BME280 sensor, so I found a linear math expression to fix the measure of sensor. The 1st expression is: 1.12 * X - 9.75, where X is the humidity. Im using BME280.py lib and my code is below. The problem is: I need to convert the value measure to new value, but I got some errors like "ValueError: invalid syntax for number". I tried to convert the humidity to float but didnt fixed the problem.

```
from machine import Pin, I2C
from time import sleep
import BME280

i2c = I2C(scl=Pin(22), sda=Pin(21), freq=10000)
i2c2 = I2C(scl=Pin(23), sda=Pin(15), freq=10000)

while True:
    try:
        bme = BME280.BME280(i2c=i2c)
        temp = bme.temperature
        hum = bme.humidity
        
        #trueHumidity1 = 1.12 * float(hum) - 9.75
        
        bme2 = BME280.BME280(i2c=i2c2)
        temp2 = bme2.temperature
        hum2 = bme2.humidity
        
        #trueHumidity2 = 1.03 * float(hum2) - 1.8
      
        print("-----------------------")
        print('Temperature[1]: ', temp)
        print('Humidity[1]:    ', hum)
        print('Temperature[2]: ', temp2)
        print('Humidity[2]:    ', hum2)
        print("-----------------------")
        sleep(1)
```

**Comment by robert-hh** [comment]
You cannot use the class members temperature and humidity directly. You have to use e.g. the values property:

`temp, press, hum  = bme.values`

or

`temp, press, hum  = bme.read_compensated_data()`

Otherwise the measurement will not take place.

**Reply by pierregraciano** [reply]
When I used the code above its worked, e.g. 
```
bme = BME280.BME280(i2c=i2c)
hum = bme.humidity
```
The real problem is I dont know how to calculate this without type error:

`trueHumidity1 = 1.12 * float(hum) - 9.75`

**Reply by robert-hh** [reply]
Please check the type of hum with `print(type(hum))`. Which board & port do you use?

**Reply by robert-hh** [reply]
Which BME280.py library do you use? Please tell it's URL.

**Reply by pierregraciano** [reply]
Im using ESP32 DevToolKit, pins:
```
i2c = I2C(scl=Pin(22), sda=Pin(21), freq=10000)
i2c2 = I2C(scl=Pin(23), sda=Pin(15), freq=10000)
```
The var "hum" and "hum2" are strings: <class 'str'>

And the BME280 lib is:
https://raw.githubusercontent.com/RuiSantosdotme/ESP-MicroPython/master/code/WiFi/HTTP_Client_IFTTT_BME280/BME280.py

**Reply by robert-hh** [reply]
So it returns the humidity as string with a trailing %. Then `float(hum[:-1])` might do the trick, which skips the last character.

**Reply by pierregraciano** [reply]
Thank you, Robert! Its worked now! Have a good night!!!
